package/MDAnalysis/core/groups.py

```diff
@@ -0,0 +1,1543 @@
+        return self.ix == other.ix
+
+    def __hash__(self):
+        return hash(self.ix)
+
+    def __add__(self, other):
+        """Concatenate the Component with another Component or Group of the
+        same level.
+
+        Parameters
+        ----------
+        other : Component or Group
+            Component or Group with `other.level` same as `self.level`
+
+        Returns
+        -------
+        Group
+            Group with elements of `self` and `other` concatenated
+        
+        """
+        if self.level != other.level:
+            raise TypeError('Can only add Atoms or AtomGroups (not "{0}")'
+                            ' to Atom'.format(other.__class__.__name__))
+
+        if not self.universe is other.universe:
+            raise ValueError("Can only add objects from the same Universe")
+
+        if isinstance(other.ix, int):
+            o_ix = np.array([other.ix])
+        else:
+            o_ix = other.ix
+
+        return self.level.plural(
+                np.concatenate((np.array([self.ix]), o_ix)), self.universe)
+
+    def __radd__(self, other):
+        """Using built-in sum requires supporting 0 + self. If other is
+        anything other 0, an exception will be raised.
+
+        Parameters
+        ----------
+        other : int
+            Other should be 0, or else an exception will be raised.
+
+        Returns
+        -------
+        self
+            Group with elements of `self` reproduced
+
+        """
+        if other == 0:
+            return self.level.plural(np.array([self._ix]), self._u)
+        else:
+            raise TypeError("unsupported operand type(s) for +:"+
+                            " '{}' and '{}'".format(type(self).__name__,
+                                                    type(other).__name__))
+
+    @classmethod
+    def _add_prop(cls, attr):
+        """Add attr into the namespace for this class
+
+        Arguments
+        ---------
+        attr 
+            TopologyAttr object to add
+        """
+        getter = lambda self: attr.__getitem__(self)
+        setter = lambda self, values: attr.__setitem__(self, values)
+
+        setattr(cls, attr.singular,
+                property(getter, setter, None, attr.singledoc))
+
+    @property
+    def universe(self):
+        return self._u
+
+    @property
+    def ix(self):
+        """Unique index of this component.
+
+        If this component is an Atom, this is the index of the atom.
+        If it is a Residue, this is the index of the residue.
+        If it is a Segment, this is the index of the segment.
+
+        """
+        return self._ix
+
+
+class Atom(ComponentBase):
+    """Atom base class.
+
+    This class is used by a Universe for generating its Topology-specific Atom
+    class. All the TopologyAttr components are obtained from ComponentBase, so
+    this class only includes ad-hoc methods specific to Atoms.
+
+    """
+    level = 'atom'
```

Comment:
As I read the code I see here a type of `str` for level. Later in this class stuff like `self.level.parent.singular` appears and I'm starting to scratch my head because a string doesn't have an attribute called `parent`. Introspection with ipython shows me that `level` at runtime is actually of type `mda.core.level.Level`. I've tried to locate where this is changed as a type from string to Level but couldn't find the code piece.

Anyway this type of class modifications during runtime make code very complex and hard to understand.  The actual interface of a variable shouldn't change from the source code definition to the runtime value. If this variable is of type Level it should be declared as such in the code itself already.